src/extension_commands.ml

```diff
@@ -96,6 +96,92 @@ let _upgrade_ocaml_lsp_server =
   command Extension_consts.Commands.upgrade_ocaml_lsp_server handler
 ;;
 
+let _install_dune_lsp_server =
+  let handler (instance : Extension_instance.t) ~args:_ =
+    let open Promise.Syntax in
+    let _ =
+      let sandbox = Extension_instance.sandbox instance in
+      match sandbox with
+      | Dune dune ->
+        let* is_dune_locked = Dune.detect_dune_lock_dir dune in
+        if is_dune_locked
+        then
+          let* dune_lsp_present = Dune.detect_dune_ocamllsp dune in
+          if dune_lsp_present
+          then
+            show_message `Info "OCaml-LSP server is already installed." |> Promise.return
+          else (
+            let options =
+              ProgressOptions.create
+                ~location:(`ProgressLocation Notification)
+                ~title:"Running `dune tools exec ocamllsp`"
+                ~cancellable:false
+                ()
+            in
+            let task ~progress:_ ~token:_ =
+              let+ result =
+                (* We first check the version so that the process can exit, otherwise the progress indicator runs \
+                   forever as the dune tools exec command automatically starts the server hence the process never ends.*)
+                Dune.exec dune ~args:[ "tools"; "exec"; "ocamllsp"; "--version" ]
+                |> Cmd.output ~cwd:(Dune.root dune)
+              in
+              match result with
+              | Ok _ -> Ojs.null
+              | Error err ->
+                show_message
+                  `Error
+                  "An error occured while running `dune tools exec ocamllsp`: %s"
+                  err;
+                Ojs.null
+            in
+            let+ _ = Vscode.Window.withProgress (module Ojs) ~options ~task in
+            let _ =
+              Dune.exec dune ~args:[ "tools"; "exec"; "ocamllsp" ]
+              |> Cmd.output ~cwd:(Dune.root dune)
+            in
+            let _ = Extension_instance.start_language_server instance in
```

Comment:
```suggestion
            let _ = Extension_instance.start_language_server instance in
```

I don't think you need to spawn the server manually here. IIUC, the `start_langauge_server` function will create a language client, which will spawn the srver as a child process. Let me know if you understand this differently than me. I'm new to this code base as well.